### DISCUSSION TITLE
Importing the testclient in fastapi==0.112.0 throws the error: ModuleNotFoundError: No module named 'httpx'

### DISCUSSION THREAD

**Discussion by pedrospinosa** [fastapi/fastapi#11958] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.testclient import TestClient
```

### Description

Example using `fastapi==0.112.0`

```python
from fastapi.testclient import TestClient 
```

Result:
```
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/py310/lib/python3.10/site-packages/starlette/testclient.py", line 33, in <module>
    import httpx
ModuleNotFoundError: No module named 'httpx'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/Caskroom/miniconda/base/envs/py310/lib/python3.10/site-packages/fastapi/testclient.py", line 1, in <module>
    from starlette.testclient import TestClient as TestClient  # noqa
  File "/opt/homebrew/Caskroom/miniconda/base/envs/py310/lib/python3.10/site-packages/starlette/testclient.py", line 35, in <module>
    raise RuntimeError(
RuntimeError: The starlette.testclient module requires the httpx package to be installed.
You can install this with:
    $ pip install httpx
```

This import works on version 0.111.1

### Operating System

macOS

### Operating System Details

macOS with M1 chip

### FastAPI Version

0.112.0

### Pydantic Version

2.8.2

### Python Version

Python 3.10.14

**Comment by bertomaniac** [comment]
You need to install it. It's mentioned in the docs when referring to the TestClient:
https://fastapi.tiangolo.com/tutorial/testing/?h=test#using-testclient

**Reply by bertomaniac** [reply, marked as answer]
I also just now realized you mentioned it works on `0.111.1` 🤦 Perhaps just try installing again? `pip install fastapi[standard]`?

**Comment by ceb10n** [comment]
The latest release was all about removing dependencies: https://github.com/fastapi/fastapi/releases/tag/0.112.0

As @tiangolo stated:

> Breaking Changes
> ♻️ Add support for pip install "fastapi[standard]" with standard dependencies and python -m fastapi. PR https://github.com/fastapi/fastapi/pull/11935 by [@tiangolo](https://github.com/tiangolo).
> 

you can see the differences in practice with `pip freeze`:

pip freeze from a virtualenv with only fastapi `0.111.1` installed:

```shell
annotated-types==0.7.0
anyio==4.4.0
certifi==2024.7.4
click==8.1.7
colorama==0.4.6
dnspython==2.6.1
email_validator==2.2.0
fastapi==0.111.1
fastapi-cli==0.0.5
h11==0.14.0
httpcore==1.0.5
httptools==0.6.1
httpx==0.27.0
idna==3.7
Jinja2==3.1.4
markdown-it-py==3.0.0
MarkupSafe==2.1.5
mdurl==0.1.2
pydantic==2.8.2
pydantic_core==2.20.1
Pygments==2.18.0
python-dotenv==1.0.1
python-multipart==0.0.9
PyYAML==6.0.2
rich==13.7.1
shellingham==1.5.4
sniffio==1.3.1
starlette==0.37.2
typer==0.12.3
typing_extensions==4.12.2
uvicorn==0.30.5
watchfiles==0.22.0
websockets==12.0
```

pip freeze from fastapi `0.112.0`:

```shell
annotated-types==0.7.0
anyio==4.4.0
fastapi==0.112.0
idna==3.7
pydantic==2.8.2
pydantic_core==2.20.1
sniffio==1.3.1
starlette==0.37.2
typing_extensions==4.12.2
```

As we can see, now we don't have `uvicorn`, `websockets`, `httpx`, `jinja2`, etc. by default. If you want all dependencies, you need to install it with `fastapi[standard]`.

pip freeze from fastapi[standard]:

```shell
annotated-types==0.7.0
anyio==4.4.0
certifi==2024.7.4
click==8.1.7
colorama==0.4.6
dnspython==2.6.1
email_validator==2.2.0
fastapi==0.112.0
fastapi-cli==0.0.5
h11==0.14.0
httpcore==1.0.5
httptools==0.6.1
httpx==0.27.0
idna==3.7
Jinja2==3.1.4
markdown-it-py==3.0.0
MarkupSafe==2.1.5
mdurl==0.1.2
pydantic==2.8.2
pydantic_core==2.20.1
Pygments==2.18.0
python-dotenv==1.0.1
python-multipart==0.0.9
PyYAML==6.0.2
rich==13.7.1
shellingham==1.5.4
sniffio==1.3.1
starlette==0.37.2
typer==0.12.3
typing_extensions==4.12.2
uvicorn==0.30.5
watchfiles==0.22.0
websockets==12.0
```

**Comment by pedrospinosa** [comment]
Oh, I see. Thanks, @bertomaniac and @ceb10n for the help!
